qemu/tests/qemu_disk_img_rebase.py

```diff
@@ -68,51 +68,68 @@ def run(test, params, env):
+        base_image = params.get("images", "image1").split()[0]
+        params_bak = copy.deepcopy(params)
+        md5_dict = {}
+        params.update(
+            {"image_name_%s" % base_image: params["image_name"],
+             "image_format_%s" % base_image: params["image_format"]})
+        image_chain = params.get("image_chain", "").split()
+        for idx, tag in enumerate(image_chain):
+            params["image_chain"] = " ".join(image_chain[:idx + 1])
+            rebase_test = RebaseTest(test, params, env, tag)
+            n_params = rebase_test.create_snapshot()
+            rebase_test.start_vm(n_params)
+            t_file = params["guest_file_name_%s" % tag]
+            md5 = rebase_test.save_file(t_file)
+            if not md5:
+                raise error.TestError("Fail to save tmp file")
+            md5_dict[t_file] = md5
+            rebase_test.destroy_vm()
 
-    # rebase snapshot image
-    rebase_chain = params.get("rebase_list", "").split(";")
-    for images in rebase_chain:
-        images = map(lambda x: x.strip(), images.split(">"))
-        try:
-            image = images[0]
-            base = images[1]
-        except IndexError:
-            msg = "Invalid format of'rebase_chain' params \n"
-            msg += "format like: 'image > base;image> base2;image2 > base2'"
-            raise error.TestError(msg)
-        params["image_chain"] = " ".join([base, image])
-        params["base_image_filename"] = image
+        # rebase snapshot image
+        rebase_chain = params.get("rebase_list", "").split(";")
+        for images in rebase_chain:
+            images = map(lambda x: x.strip(), images.split(">"))
+            try:
+                image = images[0]
+                base = images[1]
+            except IndexError:
+                msg = "Invalid format of'rebase_chain' params \n"
+                msg += "format like: 'image > base;image> base2;image2 > base2'"
+                raise error.TestError(msg)
+            params["image_chain"] = " ".join([base, image])
+            params["base_image_filename"] = image
 
-        rebase_test = RebaseTest(test, params, env, image)
-        t_params = rebase_test.rebase()
-        rebase_test.check_backingfile()
-        rebase_test.start_vm(t_params)
-        check_files = params.get("check_files", "").split()
-        for _file in check_files:
-            ret = rebase_test.check_file(_file, md5_dict[_file])
-            if not ret:
-                raise error.TestError("Check md5sum fail (file:%s)" % _file)
-        rebase_test.destroy_vm()
-        rebase_test.check_image()
+            rebase_test = RebaseTest(test, params, env, image)
+            t_params = rebase_test.rebase()
+            rebase_test.check_backingfile()
+            rebase_test.start_vm(t_params)
+            check_files = params.get("check_files", "").split()
+            for _file in check_files:
+                ret = rebase_test.check_file(_file, md5_dict[_file])
```

Comment:
the description of primary steps have been added, such as create snapshot, start guest, save temp file in guest, do the rebase test and re-check the md5 of temp file in the guest.